parachains/runtimes/bridge-hubs/test-utils/src/test_cases.rs

```diff
@@ -392,123 +347,587 @@ pub fn message_dispatch_routing_works<
+	+ pallet_bridge_relayers::Config,
+	GPI: 'static,
+	PPI: 'static,
+	MPI: 'static,
+	MB: MessageBridge,
+	<MB as MessageBridge>::BridgedChain: Send + Sync + 'static,
+	UnderlyingChainOf<MessageBridgedChain<MB>>: bp_runtime::Chain<Hash = ParaHash> + Parachain,
+	XcmConfig: xcm_executor::Config,
+	HrmpChannelOpener: frame_support::inherent::ProvideInherent<
+		Call = cumulus_pallet_parachain_system::Call<Runtime>,
+	>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	<<Runtime as pallet_bridge_messages::Config<MPI>>::SourceHeaderChain as SourceHeaderChain>::MessagesProof: From<FromBridgedChainMessagesProof<ParaHash>>,
+	<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash: From<ParaHash>,
+	ParaHash: From<<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash>,
+	<Runtime as frame_system::Config>::AccountId: From<sp_core::sr25519::Public>,
+	<Runtime as pallet_bridge_messages::Config<MPI>>::InboundRelayer: From<AccountId32>,
+	<Runtime as pallet_utility::Config>::RuntimeCall:
+	From<pallet_bridge_grandpa::Call<Runtime, GPI>>
+	+ From<pallet_bridge_parachains::Call<Runtime, PPI>>
+	+ From<pallet_bridge_messages::Call<Runtime, MPI>>
+{
+	assert_ne!(runtime_para_id, sibling_parachain_id);
+	assert_ne!(runtime_para_id, bridged_para_id);
+
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_safe_xcm_version(XCM_VERSION)
+		.with_para_id(runtime_para_id.into())
+		.with_tracing()
+		.build()
+		.execute_with(|| {
+			let zero: <Runtime as frame_system::Config>::BlockNumber = 0u32.into();
+			let genesis_hash = frame_system::Pallet::<Runtime>::block_hash(zero);
+			let mut header: <Runtime as frame_system::Config>::Header =
+				bp_test_utils::test_header(1u32.into());
+			header.set_parent_hash(genesis_hash);
+			executive_init_block(&header);
+
+			mock_open_hrmp_channel::<Runtime, HrmpChannelOpener>(
+				runtime_para_id.into(),
+				sibling_parachain_id.into(),
+			);
+
+			// start with bridged chain block#0
+			let init_data = test_data::initialization_data::<Runtime, GPI>(0);
+			pallet_bridge_grandpa::Pallet::<Runtime, GPI>::initialize(
+				<Runtime as frame_system::Config>::RuntimeOrigin::root(),
+				init_data,
+			)
+			.unwrap();
+
+			// set up relayer details and proofs
+
+			let message_destination =
+				X2(GlobalConsensus(local_relay_chain_id), Parachain(sibling_parachain_id));
+			// some random numbers (checked by test)
+			let message_nonce = 1;
+			let para_header_number = 5;
+			let relay_header_number = 1;
+
+			let relayer_at_target = Bob;
+			let relayer_id_on_target: <Runtime as frame_system::Config>::AccountId =
+				relayer_at_target.public().into();
+			let relayer_at_source = Dave;
+			let relayer_id_on_source: AccountId32 = relayer_at_source.public().into();
+
+			// Drip some balance
+			drip_some_balance(&relayer_id_on_target);
```

Comment:
@acatangiu 
ExtBuilder has possibility to set balances at genesis e.g.: https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/assets/test-utils/src/test_cases.rs#L654

or there is possibility to write shared generic helper funtction e.g.: https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/assets/test-utils/src/lib.rs#L206
```
impl<Runtime: pallet_balances::Config + <what-ever-is-neded-for-account-id>
	RuntimeHelper<Runtime>
{
        fn drip_some_balance(account: &<Runtime as frame_system::Config>::AccountId) {
                use frame_support::traits::fungible::Mutate;
                let some_currency = ExistentialDeposit::get() * 100000;
                Runtime::mint_into(account, some_currency).unwrap();
        }
```

then we dont need to copy `fn drip_some_balance` to other runtimes and also we dont need callback `drip_some_balance: fn(&<Runtime as frame_system::Config>::AccountId)`

Reply:
done :+1: